File: qwen2.5-coder/sistemaDeTipos/tipos.c
#include <stdio.h>

double calculate_average_above_threshold(int prices[], int size, int threshold) {
    if (size == 0) {
        return 0;
    }

    int count = 0;
    double sum = 0;

    for (int i = 0; i < size; ++i) {
        if (prices[i] > threshold) {
            sum += prices[i];
            ++count;
        }
    }

    if (count == 0) {
        return 0;
    }

    return sum / count;
}

int main() {
    int prices[] = {10, 20, 30, 40, 50};
    int size = sizeof(prices) / sizeof(prices[0]);
    int threshold = 20;

    double average = calculate_average_above_threshold(prices, size, threshold);

    printf("El precio promedio de los productos mayores al umbral es: %.2f\n", average);

    return 0;
}

File: llama3/sistemaDeTipos/tipos.rs
fn average_price_above_threshold(prices: & Vec<f64>, threshold: f64) -> f64 {
    if prices.is_empty() || prices.iter().all(|&price| price <= threshold) {
        0.0
    } else {
        let mut sum = 0.0;
        for &price in prices.iter() {
            if price > threshold {
                sum += price;
            }
        }
        sum / (prices.iter().filter(|&&price| price > threshold).count() as f64)
    }
}

fn main() {
    let precios_corruptos = vec![10.0, 20.0, f64::NAN, 40.0, 50.0];
    let umbral = 25.0;

    let promedio = average_price_above_threshold(&precios_corruptos, umbral);

    println!("{:.2}", promedio); // 40.00
}